packages/core/src/typeschema/validation.ts

```diff
@@ -589,7 +594,17 @@ export function matchDiscriminant(
       if (!value || !sliceElement) {
         return false;
       }
-      if (matchesSpecifiedValue(value, sliceElement)) {
+      if (sliceElement.pattern) {
```

Comment:
This is a subtle but potentially breaking validation change for malformed slice definitions. Since `matchesSpecifiedValue` returns `true` by default, checks on elements not containing either `.fixed` nor `.pattern` now return `false` whereas they used to return `true`. The only time fixed and pattern shouldn't be defined is when we're dealing with slicing by ValueSet which is handled below and retains the permissiveness for now.

Reply:
@mattwiller  - can you comment on this?  Seems ok to me, but this is kinda out of my comfort zone.

Reply:
Yep, Matt and I discussed this and I think this change should be fine.  In general, the possibility of malformed profile `StructureDefinition` resources makes validation tricky unless we assume the underlying profile is valid.  Eventually, we may want to add explicit validation checks on the profile before using it for validation of another resource: at least then we could abort and display the issue with the profile